conn.go

```diff
@@ -31,26 +49,41 @@ var (
 )
 
 // NewConn returns a new Conn using conn for I/O.
-func NewConn(conn io.ReadWriteCloser) *Conn {
+func NewConn(conn ReadWriteCloserTimeout) *Conn {
```

Comment:
Any reason why not using a `net.Conn` here instead?

Reply:
@JensRantil my thinking behind this was that to try and minimise the size of the interface required here, and only increase it by the size needed.

However if we expand this to support general deadline, writedeadline then it would be closer to https://golang.org/pkg/net/#Conn although arguable other functions in that interface aren't needed currently (like the local and remote addresses).

Reply:
Ah, makes sense. Agree with you using an interface of our own makes sense (since the local and remote addresses aren't needed). 👌